tests/TestCase/Controller/Component/FlashComponentTest.php

```diff
@@ -273,6 +273,48 @@ public function testCall(): void
         $this->assertEquals($expected, $result);
     }
 
+    /**
+     * Test magic call method, with named parameters.
+     */
+    public function testCallWithNamedParams(): void
+    {
+        $this->assertNull($this->Session->read('Flash.flash'));
+
+        $this->Flash->success(message: 'It worked');
+        $expected = [
+            [
+                'message' => 'It worked',
+                'key' => 'flash',
+                'element' => 'flash/success',
+                'params' => [],
+            ],
+        ];
+        $result = $this->Session->read('Flash.flash');
+        $this->assertEquals($expected, $result);
+
+        $this->Flash->error(message: 'It did not work', options: ['element' => 'error_thing']);
+
+        $expected[] = [
+            'message' => 'It did not work',
+            'key' => 'flash',
+            'element' => 'flash/error',
+            'params' => [],
+        ];
+        $result = $this->Session->read('Flash.flash');
+        $this->assertEquals($expected, $result, 'Element is ignored in magic call.');
+
+        $this->Flash->success(message: 'It worked', options: ['plugin' => 'MyPlugin']);
```

Comment:
Add a test for usage like `->success('message', options: ['plugin' => 'MyPlugin'])`